packages/cli/src/prompts/prompt.ts

```diff
@@ -170,13 +170,13 @@ export function argsList(contractFullName: string, methodIdentifier: string, opt
   } else return [];
 }
 
-function contractMethods(contractFullName: string, options: MethodOptions, packageFile: ZosPackageFile): any[] {
+function contractMethods(contractFullName: string, constant: boolean = false, packageFile: ZosPackageFile): any[] {
   const { contract: contractAlias, package: packageName } = fromContractFullName(contractFullName);
   const contractManager = new ContractManager(packageFile);
   if (!contractManager.hasContract(packageName, contractAlias)) return [];
   const contract = contractManager.getContractClass(packageName, contractAlias);
 
-  return contractMethodsFromAst(contract, options);
+  return constant ? contractMethodsFromAbi(contract, constant) : contractMethodsFromAst(contract, constant);
```

Comment:
:+1:

Reply:
Y has some doubts regarding this change. I loved that it was super easy to build the array of functions using the ABI, and I was wondering whether we should wrap the `contractMethodsFromAst` call inside a `try-catch`block  and use the `contractMethodsFromAbi` if the former fails (The `No AST nodes with id X found` came to my mind)